kernel-rs/src/proc.rs

```diff
@@ -732,27 +727,64 @@ impl ProcsBuilder {
     }
 
     /// Initialize the proc table at boot time.
-    pub fn init(self: Pin<&'static mut Self>) {
-        // Safe since we don't move the `Procs`.
+    pub fn init(self: Pin<&'static mut Self>) -> Pin<&'static mut Procs> {
+        // Safe: we don't move the `Procs`.
```

Comment:
혹시 `Safe:` 라고 남기는게 더 좋을까요? 이 convention에 대해 이야기할 기회가 있음 좋겠습니다

Reply:
안전성 관련 주석은 세 종류가 있습니다.
* unsafe 함수/메서드/트레잇의 safety condition 설명
  * Rust 표준 라이브러리의 convention(예: <https://doc.rust-lang.org/src/core/cell.rs.html#980-1006>)에 따라 다음과 같은 형태로 서술하고 있습니다.
```rust
/// # Safety
///
/// ...
```
* internal safety invariant가 있는 타입의 invariant 설명
  * <https://github.com/kaist-cp/rv6/pull/334#discussion_r558818027>에서 논의한 것에 따라 다음과 같은 형태로 서술하고 있습니다. 참고로, Rust 표준 라이브러리는 `// internal safety invariant: ... ` 등으로 서술합니다. (<https://doc.rust-lang.org/src/alloc/collections/vec_deque/iter_mut.rs.html#14-15>)
```rust
/// # Safety
///
/// ...
```
* unsafe 블록이 안전한 이유 설명
  * 이전에 합의한 바는 없었고 `// It's safe becuase ...`, `// Safe since ...` `// Safe: ...` 등으로 통일된 방법 없이 적고 있었습니다. Rust 표준 라이브러리의 convention(<https://doc.rust-lang.org/src/core/cell.rs.html#897-898>)에 따라 `// SAFETY: ...` 형태로 모두 통일하면 될 것 같습니다.

Reply:
unsafe 블록의 주석을 `// SAFETY: ...`로 통일하는 것은 별도 이슈로 만들어서 진행하면 어떨까요?